### DISCUSSION TITLE
Codes für Bahnhöfe ohne/mit unbekanntem UIC-Code/RIL100/länderspezifischen Code

### DISCUSSION THREAD

**Discussion by C1710** [marhei/TrainCompany-Data#488] [General]
Gerade in Osteuropa sind zumindest die mir bekannten Datenquellen eher dürftig, was Bahnhöfe angeht, es gibt keine RIL/DS100-Codes dazu, auch die UIC-Codes sind auf github.com/trainline-eu/stations nicht zu finden.

Daher wäre mein Vorschlag, sofern es keine andere öffentlich auffindbare Quelle gibt, selbst Codes zu generieren.
Um eine Kompatibilität mit allen Tools usw. zu erhalten (cc @f2k1de ?) sollte es ein einheitliches Verfahren geben.

Es gäbe aus meiner Sicht zwei mögliche Grundlagen für generierte Codes: Bahnhofsname oder der Standort.
Das Problem: Beides kann ungenau sein.
Der Standort kann durchaus durch einige Meter abweichen, bzw. ungenau sein (teils nehme ich z.B: auch einfach die Koordinaten des Ortes.).
Der Name des Bahnhofs kann auch verschiedene Schreibweisen haben, mit/ohne Bindestriche, als ASCII oder korrekt, abgekürzt vs. ausgeschrieben usw.

Andere mögliche Alternativen als Codes (nicht sicher, ob/wie (einfach) automatisiert nutzbar):
- Wikidata-Identifier - kann auch unvollständig sein
- Codes der jeweiligen Orte (es wird wahrscheinlich im Ausland max. einen Bahnhof je Stadt/Gemeinde geben)
  - Bspw. Telefon-Vorwahl
  - Problem: Auch hier kann es sein, dass die Daten nicht so leicht auffindbar sind

Noch eine Alternative: Einfach weiterhin auf Bahnhöfe mit bekannten Codes beschränken.

**Comment by f2k1de** [comment]
Für 🇨🇿 habe ich einfach die IDs aus der Stationssuche von der Webseite genommen. https://www.cd.cz/stanice
Für 🇵🇱 habe ich die IDs aus der Portal Pasażera App genommen. Kann dazu aber gerne noch einen Konverter Stationsname -> x und y Koordinate anbieten. https://play.google.com/store/apps/details?id=pl.plksa.portalpasazera

**Reply by f2k1de** [reply]
Für noch weiter Östlich müsste ich mal schauen wo man möglichst flächendeckende Daten auftreibt, weil irgendwie müssen die Bahnen ja auch ihre Bahnhöfe in ihrer IT (sofern sie welche haben) identifizieren können.

**Reply by C1710** [reply]
> Für noch weiter Östlich müsste ich mal schauen wo man möglichst flächendeckende Daten auftreibt, weil irgendwie müssen die Bahnen ja auch ihre Bahnhöfe in ihrer IT (sofern sie welche haben) identifizieren können. 

Grundsätzlich gibt es ja wahrscheinlich für alles zumindest in Europa UIC-Codes. Nur ist die Datenbank dazu nicht öffentlich zugänglich.
Standortdaten sind zumindest für mich nicht das große Problem.

**Reply by f2k1de** [reply]
Ja und da ist das Problem hat dass auch OSM oft nicht vollständig ist und fehlende Daten aufweist

**Reply by C1710** [reply]
Vielleicht ließen sich die osm_ids wenigstens da verwenden, wo sie vorhanden sind 🤔

Also, dass dann der Code nach dem Format `<Flaggen-Emoji>O<osm_id>` wäre.
Aufgrund der Länge glaube ich nicht, dass es da zu Kollisionen mit lokalen Formaten kommen kann...

**Reply by C1710** [reply]
Wo keine Haltestelle vorhanden ist, ließe sich als Fallback die ID des jeweiligen Ortes verwenden, der mit sehr großer Wahrscheinlichkeit vorhanden ist.
